src/library_fs.js

```diff
@@ -21,14 +21,15 @@ mergeInto(LibraryManager.library, {
     '$ERRNO_MESSAGES', '$ERRNO_CODES',
 #endif
     'stdin', 'stdout', 'stderr'],
-  $FS__postset: 'FS.staticInit();' +
-#if !MINIMAL_RUNTIME // MINIMAL_RUNTIME does not have __ATINIT__/__ATMAIN__/__ATEXIT__
-                '__ATINIT__.unshift(function() { if (!Module["noFSInit"] && !FS.init.initialized) FS.init() });' +
-                '__ATMAIN__.push(function() { FS.ignorePermissions = false });' +
-                '__ATEXIT__.push(function() { FS.quit() });' +
-#endif
-                // Get module methods from settings
-                '{{{ EXPORTED_RUNTIME_METHODS.filter(function(func) { return func.substr(0, 3) === 'FS_' }).map(function(func){return 'Module["' + func + '"] = FS.' + func.substr(3) + ";"}).reduce(function(str, func){return str + func;}, '') }}}',
+  $FS__postset: function() {
+    // TODO: do we need noFSInit?
+    addStaticAtInit('if (!Module["noFSInit"] && !FS.init.initialized) FS.init();');
+    addStaticAtMain('FS.ignorePermissions = false;');
+    addStaticAtExit('FS.quit();');
```

Comment:
I wonder if saying `addStaticAtInit()` is particularly illustrative, note how there's `FS.staticInit();` also below, with two different meanings of static(?). Perhaps `addRuntimeAtInit()` or just `addAtInit()`? Even though the registration is evaluated statically/at compile time, there is nothing much static about these - the functions are still run at runtime, and not at compile time.

(`addStaticAtInit()` is also ok if no better name)

Reply:
Hmm, yeah, maybe "static" means sometihng different in different cases. I intended static to mean we decide about it at compile time. But yeah, we run it at runtime. So maybe it's not the best name. Changing to "addAtInit", thanks.